src/js/jwt/insights/offline.js

```diff
@@ -50,18 +50,22 @@ export function getOfflineToken(realm, clientId) {
     return Promise.reject('not available');
   }
 
-  const tokenURL = `${insightsUrl(DEFAULT_ROUTES)}/realms/${realm}/protocol/openid-connect/token`;
+  const ssoUrl = await insightsUrl(DEFAULT_ROUTES);
+
+  const tokenURL = `${ssoUrl}/realms/${realm}/protocol/openid-connect/token`;
   const params = parseHashString(postbackUrl);
 
-  return axios({
-    method: 'post',
-    url: tokenURL,
-    data: getPostDataString(getPostDataObject(postbackUrl, clientId, params.code)),
-    config: { headers: { 'Content-Type': 'application/x-www-form-urlencoded' } },
-  }).then((response) => {
-    priv.response = response;
-    return response;
-  });
+  return () => {
```

Comment:
Are you sure about the double `return`?

Reply:
Ah, yeah! We don't have to have function body, we can just return arrow function.

Reply:
I am more curious about why we now return a function instead of a promise directly.

Reply:
Previously it was just a function that triggered async call, now it's an async function that returns a function that triggers promise. The biggest problem I had was with tests tbh. I was trying to wait for the function to finish, but since the axios never really finishes the tests were failing. This was the problematic part https://github.com/RedHatInsights/insights-chrome/pull/1742/files#diff-c3e05911df83f21e072e4d2ec2eb95fe152cf7fde3152926822a14fd0f33e558R44. Without the init to finish the axios is never fired.

There are no changes to the public API, this returned promise as well, now it returns promise as well.

Reply:
Hmm. You could mock the whole `axios` module and its implementation in the tests to fix it. I don't necessarily like doing extra magic just to fix the tests. I can easily confuse anyone who is reading the code and does not have the context.